javaslang/src-gen/main/java/javaslang/control/Applicative.java

```diff
@@ -0,0 +1,437 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+/*-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-*\
+   G E N E R A T O R   C R A F T E D
+\*-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-*/
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import javaslang.collection.List;
+import javaslang.concurrent.Future;
+
+import javaslang.Function3;
+import javaslang.Function4;
+import javaslang.Function5;
+import javaslang.Function6;
+import javaslang.Function7;
+import javaslang.Function8;
+import javaslang.control.Option;
+import javaslang.control.Try;
+import javaslang.concurrent.Future;
+
+/**
+ * Applicative helpers
+ *
+ * @author Daniel Dietrich
+ * @since 2.1.0
+ */
+public final class Applicative {
+
+    private Applicative() {}
+
+    /**
+     * Lift a function: promote it to operate on {@link Option} functors
```

Comment:
I think the terminology is "applicative functor" eg a functor containing a function. But I'm not super strong on the terminology, I was taking inspiration from the Learn You a Haskell for Great Good book:
http://learnyouahaskell.com/functors-applicative-functors-and-monoids

> When we look at it like this, we can say that liftA2 takes a normal binary function and promotes it to a function that operates on two functors.

But we can change the comment